### ISSUE TITLE
On the state of StateTree and the emergence of ActorID's

### ISSUE THREAD

**Issue by icorderi** [filecoin-project/specs#613]
- Why are there **two** maps on the `StateTree`?
- Why is the actor state map `Address -> ActorState` instead of `ActorID -> ActorState`  

```go
// current StateTree on the spec
type StateTree struct {
    SystemActors              {ActorName: addr.Address}
    ActorStates               {addr.Address: actor.ActorState}
}
```

```go
// proposed
type StateTree struct {
    ActorStates               {actor.ActorID: actor.ActorState}
}
```

**Comment by jbenet** [comment]
Yes, I believe we are getting rid of `SystemActors`. Not sure why it's there. I'll dig into it and ask, then if good, will PR up a fix later this week.

**Comment by jbenet** [comment]
Re using `actor.ActorID` to key `ActorStates` -- I would be for this as well, but let me verify w/ other impls. We're finishing some VM things this week and will hit this then

**Comment by icorderi** [comment]
Thanks @jbenet, sounds good. 
Can you link the PR for the "VM things" work to take an early peek on the direction it is heading?

**Comment by anorth** [comment]
I believe the address in `ActorStates` is the ID-address (it's confusing to distinguish between the ID integer, ID-type address, and other address types). I think the implementers agree about this.
